addons/mail/static/src/core/common/data_response_model.js

```diff
@@ -0,0 +1,65 @@
+import { Record } from "@mail/core/common/record";
+
+import { Deferred } from "@web/core/utils/concurrency";
+
+/**
+ * This class represents a specific and unique request coming from the client to the server, and it
+ * also holds the corresponding response coming from the server.
+ * It is useful when batching requests together to determine which data to return for a specific
+ * request. It basically does the un-batching by referencing the given id. The rest of the data
+ * being grouped together in the store, flattened and with no duplicate records/fields, no matter
+ * how many requests are returning the same data.
+ * Instances of the class are created by the fetch method on each call, and they are deleted once
+ * they are resolved with their data. This class should not be used directly under typical use.
+ */
+export class DataResponse extends Record {
+    static id = "id";
+    static _nextId = 0;
+
+    static createRequest() {
+        return this.insert({ id: ++this._nextId });
+    }
```

Comment:
Could be simply named `create()`.

Could even been a basic feature of JS model, to create a record. Some models like `DataResponse` auto-allocate a new record. Let's not consider for later.